tests/test-mongoc-uri.c

```diff
@@ -567,27 +567,19 @@ test_mongoc_uri_read_prefs (void)
    const read_prefs_test *t;
    int i;
 
-   bson_t *tags_dcny;
-   bson_t *tags_dcny_empty;
-   bson_t *tags_dcnyusessd_dcsf_empty;
-   bson_t *tags_empty;
-
-   tags_dcny = BCON_NEW(
+   bson_t *tags_dcny = BCON_NEW(
       "0", "{", "dc", "ny", "}"
    );
-
-   tags_dcny_empty = BCON_NEW(
+   bson_t *tags_dcny_empty = BCON_NEW(
       "0", "{", "dc", "ny", "}",
       "1", "{", "}"
    );
-
-   tags_dcnyusessd_dcsf_empty = BCON_NEW(
+   bson_t *tags_dcnyusessd_dcsf_empty = BCON_NEW(
       "0", "{", "dc", "ny", "use", "ssd", "}",
       "1", "{", "dc", "sf", "}",
       "2", "{", "}"
    );
-
-   tags_empty = BCON_NEW(
+   bson_t *tags_empty = BCON_NEW(
```

Comment:
This code doesn't compile using MSVS 2010 see https://jira.mongodb.org/browse/CDRIVER-981?jql=project%20%3D%20CDRIVER%20AND%20resolution%20%3D%20Unresolved%20ORDER%20BY%20priority%20DESC
MSVS doesn't support c99 standard(only partial support). ansi c standard requires variable definition at the beginning of the scope.